documentation/CAMARA-API-Design-Guide.md

```diff
@@ -0,0 +1,1165 @@
+
+In notification scenarios (i.e., POST request sent towards the listener indicated by `sink` address),
+the use of the "x-correlator" is supported for the same aim as well.
+When the API request includes the "x-correlator" header,
+it is recommended for the listener to include it in the response with the same value as was used in the request.
+Otherwise, it is optional to include the "x-correlator" header in the response with any valid value.
+
+NOTE: HTTP headers are case-insensitive. The use of the naming `x-correlator` is a guideline to align the format across CAMARA APIs. 
+
+##### Content-Type Header - clarification
+
+The character set supported must only be UTF-8. The [JSON Data Interchange Format (RFC 8259)](https://datatracker.ietf.org/doc/html/rfc8259#section-8.1) states the following
+
+```
+JSON text exchanged between systems that are not part of a closed
+ecosystem MUST be encoded using UTF-8 [RFC3629].
+
+Previous specifications of JSON have not required the use of UTF-8
+when transmitting JSON text.  However, the vast majority of JSON-
+based software implementations have chosen to use the UTF-8 encoding,
+to the extent that it is the only encoding that achieves
+interoperability.
+```
+
+Implementers may include the UTF-8 character set in the Content-Type header value, for example, "`application/json; charset=utf-8`". However, the preferred format is to specify only the MIME type, such as "`application/json`". Regardless, the interpretation of the MIME type as UTF-8 is mandatory, even when only "`application/json`" is provided.
+
+#### 5.8.6. Security schemes
+
+[Security schemes](https://spec.openapis.org/oas/v3.0.3#security-scheme-object) express security in OpenAPI. 
+Security can be expressed for the API as a whole or for each endpoint.
+
+As specified in [Use of openIdConnect for securitySchemes](https://github.com/camaraproject/IdentityAndConsentManagement/blob/main/documentation/CAMARA-API-access-and-user-consent.md#use-of-openidconnect-for-securityschemes), all CAMARA OpenAPI files must include the following scheme definition, with an adapted `openIdConnectUrl` in its components section. The schema definition is repeated in this document for illustration purposes, the correct format must be extracted from the link above.
+
+```yaml
+components:
+  securitySchemes:
+    openId:
+      type: openIdConnect
+      openIdConnectUrl: https://example.com/.well-known/openid-configuration
+```
+
+The key of the security scheme is arbitrary in OAS, but convention in CAMARA is to name it `openId`.
+
+## 6. Security
+
+### 6.1. Good practices to secure REST APIs
+
+The following points can serve as a checklist to design the security mechanism of the REST APIs.

```

Comment:
These recommendations are a blend of 'security by design' (which can be applied by the API authors during API definition) and security at the implementation phase. If we are covering both. I suggest we split them into two sections to account for what the CAMARA API authors can define, and  best practice for implementers of those API definitions.